src/gtk/webview_webkit2_extension.cpp

```diff
@@ -0,0 +1,426 @@
+  "   <arg type='t' name='page_id' direction='in'/>"
+  "   <arg type='b' name='has_selection' direction='out'/>"
+  "  </method>"
+  " </interface>"
+  "</node>";
+
+class wxWebViewWebKitExtension;
+
+static gboolean
+wxgtk_webview_authorize_authenticated_peer_cb(GDBusAuthObserver *observer,
+                                              GIOStream *stream,
+                                              GCredentials *credentials,
+                                              wxWebViewWebKitExtension *extension);
+static void
+wxgtk_webview_dbus_connection_created_cb(GObject *source_object,
+                                         GAsyncResult *result,
+                                         wxWebViewWebKitExtension *extension);
+
+static wxWebViewWebKitExtension *extension = NULL;
+
+class wxWebViewWebKitExtension
+{
+public:
+    wxWebViewWebKitExtension(WebKitWebExtension *webkit_extension, const char* server_address);
+    void ClearSelection(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void DeleteSelection(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void GetPageSource(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void GetPageText(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void GetSelectedSource(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void GetSelectedText(GVariant *parameters, GDBusMethodInvocation *invocation);
+    void HasSelection(GVariant *parameters, GDBusMethodInvocation *invocation);
+
+    GDBusConnection *m_dbus_connection;
+    WebKitWebExtension *m_webkit_extension;
+
+private:
+    WebKitWebPage* GetWebPageOrReturnError(GVariant *parameters, GDBusMethodInvocation *invocation);
+
+    GCancellable *m_cancellable;
+};
+
+wxWebViewWebKitExtension::wxWebViewWebKitExtension(WebKitWebExtension *extension, const char* server_address)
+{
+    m_webkit_extension = (WebKitWebExtension*)g_object_ref(extension);
+
+    m_cancellable = g_cancellable_new();
+
+    GDBusAuthObserver *observer = g_dbus_auth_observer_new();
+    g_signal_connect(observer, "authorize-authenticated-peer",
+                     G_CALLBACK(wxgtk_webview_authorize_authenticated_peer_cb),
+                     this);
+
+    g_dbus_connection_new_for_address(server_address,
+                                      G_DBUS_CONNECTION_FLAGS_AUTHENTICATION_CLIENT,
+                                      observer,
+                                      m_cancellable,
+                                      (GAsyncReadyCallback)wxgtk_webview_dbus_connection_created_cb,
+                                      this);
+    g_object_unref (observer);
+}
+
+WebKitWebPage* wxWebViewWebKitExtension::GetWebPageOrReturnError(GVariant *parameters, GDBusMethodInvocation *invocation)
+{
+    guint64 page_id;
+    g_variant_get(parameters, "(t)", &page_id);
+    WebKitWebPage *web_page;
+    web_page = webkit_web_extension_get_page(m_webkit_extension, page_id);
```

Comment:
This is a case where I was trying to avoid exceeding 80 characters.  Obviously I didn't succeed in doing that throughout.  :(